joern-cli/src/main/scala/io/joern/joerncli/JoernExport.scala

```diff
@@ -5,35 +5,55 @@ import better.files.File
 import io.joern.dataflowengineoss.DefaultSemantics
 import io.joern.dataflowengineoss.layers.dataflows._
 import io.joern.dataflowengineoss.semanticsloader.Semantics
-import io.joern.joerncli.CpgBasedTool.{exitIfInvalid, exitWithError}
+import io.joern.joerncli.CpgBasedTool.exitIfInvalid
 import io.joern.x2cpg.layers._
+import io.shiftleft.codepropertygraph.Cpg
+import io.shiftleft.semanticcpg.language.{toAstNodeMethods, toNodeTypeStarters}
 import io.shiftleft.semanticcpg.layers._
-import overflowdb.Graph
 import overflowdb.formats.ExportResult
 import overflowdb.formats.dot.DotExporter
 import overflowdb.formats.graphml.GraphMLExporter
-import overflowdb.formats.neo4jcsv.Neo4jCsvExporter
 import overflowdb.formats.graphson.GraphSONExporter
+import overflowdb.formats.neo4jcsv.Neo4jCsvExporter
+import overflowdb.{Edge, Node}
 
-import java.nio.file.Paths
+import java.nio.file.{Path, Paths}
+import scala.jdk.CollectionConverters.IteratorHasAsScala
 import scala.util.Using
 
 object JoernExport extends App {
 
   case class Config(
     cpgFileName: String = "cpg.bin",
     outDir: String = "out",
-    repr: Representation.Value = Representation.cpg14,
-    format: Format.Value = Format.dot
+    repr: Representation.Value = Representation.Cpg14,
+    format: Format.Value = Format.Dot
   )
 
   /** Choose from either a subset of the graph, or the entire graph (all).
     */
   object Representation extends Enumeration {
-    val ast, cfg, ddg, cdg, pdg, cpg14, all = Value
+    val Ast, Cfg, Ddg, Cdg, Pdg, Cpg14, All, SplitByMethod = Value
```

Comment:
I guess it's a bit strange that ASTs, CFGs, Ddgs, ... are also split by method, and then there's a `SplitByMethod`, but I also can't think of a different way of doing this right now. I guess as long as the docs explain this, we're fine.

Reply:
discussed via slack: we'll rename `SplitByMethod` to `Cpg`
...  there's no optimal solution 🤷🏻